lectures/10-binary search/code/src/com/kunal/SortedMatrix.java

```diff
@@ -0,0 +1,82 @@
+package com.kunal;
+
+import java.util.Arrays;
+
+public class SortedMatrix {
+    public static void main(String[] args) {
+        int[][] arr = {
+                {1, 2, 3},
+                {4, 5, 6},
+                {7, 8, 9}
+        };
+        System.out.println(Arrays.toString(search(arr, 9)));
+    }
+
+    // search in the row provided between the cols provided
+    static int[] binarySearch(int[][] matrix, int row, int cStart, int cEnd, int target) {
+        while (cStart <= cEnd) {
+            int mid = cStart + (cEnd - cStart) / 2;
+            if (matrix[row][mid] == target) {
+                return new int[]{row, mid};
+            }
+            if (matrix[row][mid] < target) {
+                cStart = mid + 1;
+            } else {
+                cEnd = mid - 1;
+            }
+        }
+        return new int[]{-1, -1};
+    }
+
+    static int[] search(int[][] matrix, int target) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.